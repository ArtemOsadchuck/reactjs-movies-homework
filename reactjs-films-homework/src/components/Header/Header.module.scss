.header {
  align-items: center;
  background-color: #333;
  display: flex;
  flex-wrap: wrap;
  height: 5rem;
  justify-content: space-between;
  margin: 0;
  padding: 0% 5%;
  position: relative;
  top: 0;
  width: 100%;

  @media screen and (max-width: 760px) {
    margin: 0;
  }

  h1 {
    color: rgb(255, 255, 255);
    flex-grow: 2;
    letter-spacing: 2px;
    margin-left: 0.5rem;
    text-align: start;
  }

  a {
    color: rgb(255, 255, 255);
    text-decoration: none;
    text-decoration-color: none;
  }
}
