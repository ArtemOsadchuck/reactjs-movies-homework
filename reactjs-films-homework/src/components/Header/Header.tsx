import React from 'react';
import Form from './Form';
import LanguageToggler from './LanguageToggler';
import style from './Header.module.scss';

import lang from '../../languages/getLanguage';
import { useAppSelector } from '../../hooks/hooks';

const Header: React.FC = () => {
  const appLang = useAppSelector((state) => state.mainReducer.lang);
  return (
    <header className={style.header}>
      <h1>{lang(appLang).title.toUpperCase()}</h1>
      <Form placeholder={lang(appLang).placeholder} />
      <LanguageToggler />
    </header>
  );
};

export default Header;
