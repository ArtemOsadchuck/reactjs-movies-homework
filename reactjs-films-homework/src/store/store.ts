import rootStore from './rootStore/rootStore';

export type RootState = ReturnType<typeof rootStore.getState>;
export type AppDispatch = typeof rootStore.dispatch;
export default rootStore;
