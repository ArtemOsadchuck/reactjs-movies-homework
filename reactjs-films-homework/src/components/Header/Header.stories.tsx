import React from 'react';
import { Story } from '@storybook/react';

import Header from './Header';

import { Provider } from 'react-redux';
import store from '../../store/store';

const Template: Story<typeof Header> = () => (
  <Provider store={store}>
    <Header />
  </Provider>
);

export const HeaderStory = Template.bind({});
export default {
  title: 'Header',
  component: Header,
};
