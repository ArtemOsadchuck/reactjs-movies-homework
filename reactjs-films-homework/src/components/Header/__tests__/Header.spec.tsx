import React from 'react';
import { cleanup, render, screen } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';
import Header from '../Header';
import { Provider } from 'react-redux';
import store from '../../../store/store';

describe('Header', () => {
  let fragment: any;
  const history = createMemoryHistory();
  history.push('/');

  beforeEach(() => {
    const { asFragment } = render(
      <Provider store={store}>
        <Router history={history}>
          <Header />
        </Router>
      </Provider>
    );
    fragment = asFragment();
  });

  afterEach(() => {
    cleanup();
  });

  it('Snapshot', () => {
    expect(fragment).toMatchSnapshot();
  });

  it('Must include img', () => {
    expect(screen.getByRole('img')).toBeInTheDocument();
  });

  it('Must include h1', () => {
    expect(screen.getByRole('heading')).toBeInTheDocument();
  });

  it('Must include button', () => {
    expect(screen.getByRole('button')).toBeInTheDocument();
  });
});
