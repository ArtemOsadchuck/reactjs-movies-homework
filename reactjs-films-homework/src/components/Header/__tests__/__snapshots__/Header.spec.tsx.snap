// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Snapshot 1`] = `
<DocumentFragment>
  <header
    class="header"
  >
    <h1>
      <a
        href="/?category=popular&page=1"
      >
        TITLE
      </a>
    </h1>
    <div
      class="formWrapper"
    >
      <form
        class="form"
        id="form"
      >
        <input
          autocomplete="off"
          class="input"
          id="input_search"
          placeholder="Movies, person, movie theaters"
          type="search"
          value=""
        />
        <div
          class="search"
          id="search"
        >
          <a
            href="/?search=&page=1"
          >
            <img
              alt=""
              id="search"
              src="search.png"
              width="16px"
            />
          </a>
        </div>
      </form>
    </div>
    <div
      class="dropdownWrapper"
    >
      <button
        class="dropdownBtn"
      >
        EN
      </button>
    </div>
  </header>
</DocumentFragment>
`;
